### PR TITLE
Quruom store

### PR DESCRIPTION
**Pull request by sasha8**
QuorumStore draft. For integration tests.

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/1571)

### CODE REVIEW

File: consensus/src/epoch_manager.rs [hunk 1]
```diff
@@ -309,16 +324,116 @@ impl EpochManager {
...
+    ) -> Arc<BatchReader> {
+        let backend = &self.config.safety_rules.backend;
+        let storage: Storage = backend.try_into().expect("Unable to initialize storage");
+        if let Err(error) = storage.available() {
+            panic!("Storage is not available: {:?}", error);
+        }
+        let private_key = storage
+            .export_private_key(CONSENSUS_KEY)
+            .expect("Unable to get private key");
+        let signer = ValidatorSigner::new(self.author, private_key);
```
Comment (bchocho) [inline consensus/src/epoch_manager.rs:355]: @zekun000 #1764 doesn't update `export_private_key(...)` to use BLS. Should it be updated? Or is there another recommended way to get the private key?

Comment (bchocho) [reply consensus/src/epoch_manager.rs:355]: Changing to `storage.get(CONSENSUS_KEY)...` I see on smoke tests:

```
2022-07-15T21:57:18.787506Z [consensus] ERROR crates/crash-handler/src/lib.rs:38 details = '''panicked at 'Unable to get private key: KeyNotSet("consensus")', consensus/src/epoch_manager.rs:410:14'''
backtrace = '''
```

Comment (bchocho) [reply consensus/src/epoch_manager.rs:355]: @zekun000 I see that smoke tests now use InMemoryStorage. This breaks the assumption here that reading from storage after `backend.try_into()` can read the private key (for InMemoryStorage this just creates a fresh HashMap).

I see the configs themselves have changed in the smoke tests.

Before:
```
safety_rules: SafetyRulesConfig { backend: OnDiskStorage(OnDiskStorageConfig
```
After:
```
safety_rules: SafetyRulesConfig { backend: InMemoryStorage
```

Is this an intended change? I couldn't figure out where exactly the config changed. I guess we should either make the InMemory store a singleton so the same HashMap is used, or just revert back to OnDisk.

**Comment by davidiw** [comment]
looks like the older of two prs

**Pull request closed without merging**